### PR TITLE
[REVIEW] Create nvtext normalize_characters API from the subword_tokenize internal function

### LINKED ISSUE #5527: [FEA] Remove accents from strings
**Is your feature request related to a problem? Please describe.**
For some NLP operation is necessary to remove accents from letters, for example, transform é, è, â, î, ô to e, e, a, i, o.
In pandas, you can use `str.normalize` to do the work

**Describe the solution you'd like**
A pandas like function `str.normalize` to remove accents from strings

**Describe alternatives you've considered**
It would be possible to use regular expressions to remove accents

### PR DESCRIPTION
**Pull request by davidwendt**
Closes #5527 

This PR exposes the character normalizing logic within the subword-tokenizer function as a separate API.

```
std::unique_ptr<cudf::column> normalize_characters(
  cudf::strings_column_view const& strings,
  bool do_lower_case,
  rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
```

The `do_lower_case` allows lower-case and accent removal to be optional to match the existing feature of the internal code.

### CODE REVIEW

File: python/cudf/cudf/core/column/string.py [hunk 1]
```diff
@@ -3616,6 +3619,55 @@ def normalize_spaces(self, **kwargs):
...
+        """
+        Normalizes strings characters for tokenizing.
+
+        This uses the normalizer that is built into the
+        subword_tokenize function which includes:
+
+            - adding padding around punctuation (unicode category starts with
+              "P") as well as certain ASCII symbols like "^" and "$"
+            - adding padding around the CJK Unicode block characters
+            - changing whitespace (e.g. `"\t", "\n", "\r"`) to space `" "`
```
Comment (kkraus14) [inline python/cudf/cudf/core/column/string.py:3632]: @galipremsagar do we need to escape `\` characters for Sphinx?

**Comment by kkraus14** [comment]
LGTM for Python other than one docstring question

File: python/cudf/cudf/core/column/string.py (hunk 1 above)
Comment (davidwendt) [reply python/cudf/cudf/core/column/string.py:3632]: Looks like I need to add another `\` to these. I see this in other similar places in the file. Example:
https://github.com/rapidsai/cudf/blob/79bd43517407302f4b2e52b5b470fdfcafe51a6b/python/cudf/cudf/core/column/string.py#L963

Comment (galipremsagar) [reply python/cudf/cudf/core/column/string.py:3632]: Yep, we will need `\` for this kind of instance too inside backticks, else would result in broken docs.

Comment (galipremsagar) [reply python/cudf/cudf/core/column/string.py:3632]: Would recommend this: `(e.g. ``\\t``, ``\\n``, ``\\r``)` as having single back-tick seems to be not representing correctly in the end output for this text.

File: cpp/src/text/normalize.cu [hunk 2]
```diff
@@ -130,4 +264,12 @@ std::unique_ptr<cudf::column> normalize_spaces(cudf::strings_column_view const&
   return detail::normalize_spaces(strings, mr);
 }
 
+std::unique_ptr<cudf::column> normalize_characters(cudf::strings_column_view const& strings,
```
Comment (nvdbaranec) [inline cpp/src/text/normalize.cu:267]: Maybe throw a @copydoc here and up in the detail definition.

File: cpp/include/nvtext/normalize.hpp [hunk 3]
```diff
@@ -51,5 +51,54 @@ std::unique_ptr<cudf::column> normalize_spaces(
   cudf::strings_column_view const& strings,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Normalizes strings characters for tokenizing.
+ *
+ * This uses the normalizer that is built into the nvtext::subword_tokenize function
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/include/nvtext/normalize.hpp:57]: I wanted to make it clear that these two functions are in sync -- future requirements against the subword-tokenizer may effect future behavior of this function.

File: cpp/tests/text/normalize_tests.cpp [hunk 4]
```diff
@@ -67,7 +67,59 @@ TEST_F(TextNormalizeTest, NormalizeEmptyTest)
...
+                                                 "ACEN U",
+                                                 "P ^ NP",
+                                                 " $ 41 . 07",
+                                                 " [ a , b ] ",
+                                                 " 丏  丟 ",
+                                                 ""},
+                                                validity);
+    cudf::test::expect_columns_equal(*results, expected);
+  }
+}
```
Comment (harrism) [inline cpp/tests/text/normalize_tests.cpp:125]: I'm surprised this passes clang-format.

```suggestion

}
```

File: cpp/include/nvtext/normalize.hpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (harrism) [reply cpp/include/nvtext/normalize.hpp:57]: Seems like an implementation detail that shouldn't be in public docs, but OK.

**Comment by davidwendt** [comment]
rerun tests

**Pull request merged**